manifests/zone.pp

```diff
@@ -36,4 +38,34 @@
     replace => false,
     notify  => Service[$::dns::namedservicename],
   }
+
+  if $static_records == true {
+    concat_fragment { "dns-static-${zone}+01header.dnsstatic":
+      content => "; static file for ${zone}",
+    }
+  
+    Dns::Record <<| |>> { notify => Concat_build["dns-static-${zone}"] }
```

Comment:
I found that without the 'notify' part I needed a second run to get the records to show up.